### PR TITLE
Moving Nest Changes

### PR DESCRIPTION
**Pull request by wramstrom**
**Description**

Added new subroutine [mpp_shift_nest_domains()] and new variable [nest_level] to enable moving nests, by modifying the domain data structures.
Fixes # 838
https://github.com/NOAA-GFDL/FMS/issues/838

**How Has This Been Tested?**
Tests of C96 and C768 global configurations with 3X moving nest on tile 6 have been performed on Orion and Hera.

**Checklist:**
- [YES] My code follows the style guidelines of this project
- [YES] I have performed a self-review of my own code
- [YES] I have commented my code, particularly in hard-to-understand areas
- [YES ] I have made corresponding changes to the documentation
- [YES] My changes generate no new warnings
- [YES] Any dependent changes have been merged and published in downstream modules
- [ ] New check tests, if applicable, are included
- [ ] `make distcheck` passes

### CODE REVIEW

**Comment by rem1776** [comment]
> @wramstrom Hello, looks like this hasn't been updated in a while. When you get a chance could you make the requested changes so we can merge this in?

Just following up, do you still need these updates merged?

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,12 @@ and this project uses `yyyy.rr[.pp]`, where `yyyy` is the year a patch is releas
 `rr` is a sequential release number (starting from `01`), and an optional two-digit
 sequential patch number (starting from `01`).
 
+## [2021.03.01] - 2021-09-07
+### Fixed
+- TIME_INTERP: Fixes issue in load_record when reading 3d variables with fms2_io and elimates redundant loads and validity checks for on-grid interpolations
+### Changed
+- Changes configure script to only check for gcc 11.1.0, not any gcc 11 version
+
```
[the remark this answers is not in the document]
Comment (wramstrom) [reply CHANGELOG.md]: This has now been removed from the code.

File: mpp/include/mpp_define_nest_domains.inc [hunk 2]
```diff
@@ -358,6 +364,82 @@ subroutine mpp_define_nest_domains(nest_domain, domain, num_nest, nest_level, ti
...
 
+!  Ramstrom/HRD Moving Nest
+!  Based on mpp_define_nest_domains, but just resets positioning of nest 
+!    Modifies the parent/coarse start and end indices of the nest location
+!    Computes new overlaps of nest PEs on parent PEs
+subroutine mpp_shift_nest_domains(nest_domain, domain, delta_i_coarse, delta_j_coarse, extra_halo)
+  type(nest_domain_type),     intent(inout) :: nest_domain
+  type(domain2D), target,     intent(in   ) :: domain
+  integer,                    intent(in   ) :: delta_i_coarse(:), delta_j_coarse(:)
+  integer,          optional, intent(in   ) :: extra_halo
```
[the remark this answers is not in the document]
Comment (wramstrom) [reply mpp/include/mpp_define_nest_domains.inc]: I added Doxygen style comments to describe the subroutine and the arguments.

File: mpp/include/mpp_define_nest_domains.inc [hunk 3]
```diff
@@ -358,6 +364,82 @@ subroutine mpp_define_nest_domains(nest_domain, domain, num_nest, nest_level, ti
...
+
+  integer                                   :: n, l, m, my_tile_coarse
+  integer                                   :: num_nest
+  integer                                   :: extra_halo_local, npes_nest_top
+  integer                                   :: nnest, nlevels, ntiles_top, ntiles, pos
+  integer, pointer                          :: nest_level(:)
```
[the remark this answers is not in the document]
Comment (wramstrom) [reply mpp/include/mpp_define_nest_domains.inc]: I added Doxygen comments for the subroutine arguments.  I did not add comments for the internal variables, following the format of other subroutines in this file.

File: mpp/mpp_domains.F90 [hunk 4]
```diff
@@ -458,6 +459,7 @@ module mpp_domains_mod
   type :: nest_domain_type
      character(len=NAME_LENGTH)     :: name
      integer                        :: num_level
+     integer,               pointer :: nest_level(:)    !  Ramstrom -- Added for moving nest functionality
```
[the remark this answers is not in the document]
Comment (wramstrom) [reply mpp/mpp_domains.F90]: This comment has been reformatted into Doxygen style with developer's name removed.

**Comment by wramstrom** [comment]
> @wramstrom do you have a test that you can provide to make sure this code update works as intended in the future?

This can be tested with some of the code in the hafs_nesting branch of FV3/atmos_cubed_sphere.  I do not have straightforward unit testing available for this subroutine.

**Review by thomas-robinson — APPROVED** [review]

**Review by rem1776 — APPROVED** [review]

**Review by bensonr — COMMENTED** [review]
@wramstrom - I have some questions and would like to do a verbal code review to understand some of the logic in mpp_shift_nest_domains.  My calendar is up to date and you can feel free to schedule when it's convenient for you.

**Pull request merged**